src/LibHdb++Cassandra.cpp

```diff
@@ -276,214 +407,197 @@ bool HdbPPCassandra::find_attr_id_and_ttl(string fqdn_attr_name, CassUuid & ID,
 
-		const CassResult* result = cass_future_get_result(future);
-		CassIterator* iterator = cass_iterator_from_result(result);
-		if(cass_iterator_next(iterator))
+		if (cassTtlVal == NULL)
 		{
-#ifdef _CASS_LIB_DEBUG
-			cout << __func__<< ": SUCCESS in query: " << query_str.str() 
-			     << "(" << CONF_COL_NAME << "=\'" << attr_name << "\',"
-				 << CONF_COL_FACILITY << "=\'" << facility << "\')" << endl;
-#endif
-			const CassRow* row = cass_iterator_get_row(iterator);
-			cass_value_get_uuid(cass_row_get_column(row, 0), &ID);
-			cass_int32_t ttl_val = 0;
-#ifdef _CASS_LIB_DEBUG
-			const CassValue* cassTtlVal = cass_row_get_column_by_name(row, CONF_COL_TTL.c_str());
-			if(cassTtlVal == NULL)
-			{
-				cout << "Column ttl does not exist???" << endl;
-			}
-			else
-			{
-				cout << "OK . value_type() = " << cass_value_type(cassTtlVal) << endl;
-			}
-#endif
-			CassError get_ttl_cass_error = cass_value_get_int32(cass_row_get_column_by_name(row, CONF_COL_TTL.c_str()), &ttl_val);
-			if(get_ttl_cass_error != CASS_OK)
-			{
+			DEBUG_MSG("Column ttl does not exist???" << endl);
+			// TODO (SJ) Is this an error condition?
+		}
+		else
+		{
+			DEBUG_MSG("OK . value_type() = " << cass_value_type(cassTtlVal) << endl);
+		}
 
-#ifdef _CASS_LIB_DEBUG
-				if(get_ttl_cass_error == CASS_ERROR_LIB_NULL_VALUE)
-				{
-					cout << "TTL value is NULL" << endl;
-				}
-				cout << "An error occured: " <<  cass_error_desc(get_ttl_cass_error) << endl;
-				cout << "cass_value_get_int32 for ttl detected a NULL value" << endl;
-#endif
-				ttl = 0;
-			}
-			else
-			{
-#ifdef _CASS_LIB_DEBUG
-				cout << "cass_value_get_int32 returned " << ttl_val << endl;
-#endif
-				ttl = ttl_val;
-			}
-			struct ConfParams conf_param;
-			conf_param.id = ID;
-			conf_param.ttl = ttl;
-#ifdef _CASS_LIB_DEBUG
-			cout << __func__<< "(" << fqdn_attr_name << "): ID found " << endl;
-			cout << __func__<< "(" << fqdn_attr_name << "): TTL = " << ttl << endl;
-#endif
-			found = true;
-			attr_ID_map.insert(make_pair(fqdn_attr_name,conf_param));
-			map<string,struct ConfParams>::iterator it = attr_ID_map.find(fqdn_attr_name);
-			if(it == attr_ID_map.end())
-			{
-				char uuidStr[CASS_UUID_STRING_LENGTH];
-				cass_uuid_string(ID,uuidStr);
-				cout << "WARNING: (" << __FILE__ << "," << __LINE__ << ") "<< __func__<< ": ID (" << uuidStr << ") could not be added into cache for attr ="<< fqdn_attr_name << endl;
-			}
-    	}
-		cass_result_free(result);
-		cass_iterator_free(iterator);
+		CassError get_ttl_cass_error = cass_value_get_int32(cass_row_get_column_by_name(row, CONF_COL_TTL.c_str()), &ttl_val);
+
+		if(get_ttl_cass_error != CASS_OK)
+		{
+			if (get_ttl_cass_error == CASS_ERROR_LIB_NULL_VALUE)
+				DEBUG_MSG("TTL value is NULL" << endl);
+
+			DEBUG_MSG("An error occured: " << cass_error_desc(get_ttl_cass_error) << endl);
+			DEBUG_MSG("cass_value_get_int32 for ttl detected a NULL value" << endl);
+			// TODO (SJ) Is this an error condition?
```

Comment:
We end up with:

```
		if(get_ttl_cass_error != CASS_OK)
		{
			if (get_ttl_cass_error == CASS_ERROR_LIB_NULL_VALUE)
				DEBUG_MSG("TTL value is NULL" << endl);
			else
				DEBUG_MSG("An error occured: " << cass_error_desc(get_ttl_cass_error) << endl);

			ttl = 0;
		}
````

Is that else an error condition we should raise an exception for? It appears to be an unknown errtor state?